src/topology/noetherian_space.lean

```diff
@@ -0,0 +1,185 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.sets.compacts
+import order.order_iso_nat
+import order.compactly_generated
+
+/-!
+# Noetherian space
```

Comment:
Could you please discuss some examples of Noetherian spaces in the module docstring. Also, please prove (using an instance or a lemma) that a finite type is a Noetherian space.

Reply:
It should be easy to show the prime spectrum of a noetherian ring is a noetherian space using well_foundedness of (<) on closeds, but that should be done in https://leanprover-community.github.io/mathlib_docs/algebraic_geometry/prime_spectrum/basic.html IMO.
The following lemma together with [inv_image.wf](https://leanprover-community.github.io/mathlib_docs/init/wf.html#inv_image.wf) and [subrelation.wf](https://leanprover-community.github.io/mathlib_docs/init/wf.html#subrelation.wf) should suffice:
```
import algebraic_geometry.prime_spectrum.basic
open prime_spectrum
theorem vanishing_ideal_anti_mono_iff {R : Type*} [comm_ring R]
  {s t : set (prime_spectrum R)} (hs : is_closed s) (ht : is_closed t) :
  s ⊆ t ↔ prime_spectrum.vanishing_ideal t ≤ prime_spectrum.vanishing_ideal s :=
⟨vanishing_ideal_anti_mono, λ h,
begin
  rw [← hs.closure_eq, ← ht.closure_eq],
  convert ← zero_locus_anti_mono_ideal h;
  apply zero_locus_vanishing_ideal_eq_closure,
end⟩
```
(To convert between `≤` and `<` you need both directions of the lemma.)

Reply:
I added a sentence in the docstring mentioning the underlying space of a noetherian scheme, but I don't really know any other interesting example.